storage/innobase/api/api0api.cc

```diff
@@ -2960,7 +2960,8 @@ dberr_t ib_sdi_set(uint32_t tablespace_id, const ib_sdi_key_t *ib_sdi_key,
                                   << " Error returned: " << err
                                   << " by trx->id: " << trx->id;);
 
-    ut_ad(err == DB_SUCCESS || trx_is_interrupted(trx) || !"sdi_insert_failed");
+    ut_ad(err == DB_SUCCESS || err == DB_IO_DECRYPT_FAIL ||
+          trx_is_interrupted(trx) || !"sdi_insert_failed");
```

Comment:
preexisting of course, but !"sdi_insert_failed" - is this an attempt to provide some sort of message in the output? I am not sure if it's worthy enough to keep

Reply:
Hey its me who has added this message :) I can give the background around this.
The assert initially is about crashing when SDI operations fail. They cannot fail under normal circumstances.

But with some debug options (innodb_trx_undo_n_slots_debug or something like that), the assert here crashed. Many bugs are raised by percona (Roel). With that settings, it crashed at other places too apart from SDI operations. (IMHO, that debug setting was a very specific purpose)

The fix is tolerate such failures for those weird cases.
So that part of message shouldn't be removed. The string is to provide a unique signature for the assert instead of  regular Assertion 0 failed.